src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5587,16 +5587,40 @@ public override AstVisitAction VisitAssignmentStatement(AssignmentStatementAst a
                     return AstVisitAction.StopVisit;
                 }
 
-                if (assignmentStatementAst.Left is ConvertExpressionAst convertExpression)
+                if (assignmentStatementAst.Left is AttributedExpressionAst attributedExpression)
                 {
-                    if (convertExpression.Child is VariableExpressionAst variableExpression)
+                    var firstConvertExpression = attributedExpression as ConvertExpressionAst;
+                    ExpressionAst child = attributedExpression.Child;
+                    while (child is AttributedExpressionAst attributeChild)
+                    {
+                        if (firstConvertExpression is null && attributeChild is ConvertExpressionAst convertExpression)
+                        {
+                            // Multiple type constraint can be set on a variable like this: [int] [string] $Var1 = 1
+                            // However, only the first one is actually applied to the variable.
+                            firstConvertExpression = convertExpression;
```

Comment:
The comment is not accurate. All conversions applied from right to left. For completion we need to store most left one.

Reply:
I have updated the comment. Please check again.